compiler/luci/profile/include/luci/Profile/CircleNodeOrigin.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2021 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __LUCI_PROFILE_CIRCLE_NODE_ORIGIN_H__
+#define __LUCI_PROFILE_CIRCLE_NODE_ORIGIN_H__
+
+#include "CircleNodeID.h"
+
+#include <luci/IR/CircleNode.h>
+
+#include <set>
+
+namespace luci
+{
+
+class CircleNodeOrigin
+{
+protected:
+  struct Source
+  {
+  public:
+    std::string name(void) const { return _name; }
+    void name(const std::string &name) { _name = name; }
+
+    uint32_t id(void) const { return _id; }
+    void id(const uint32_t id) { _id = id; }
+
+  private:
+    std::string _name;
+    uint32_t _id = 0;
+  };
+
+public:
+  virtual std::set<const Source *> sources(void) const = 0;
+};
+
+std::shared_ptr<CircleNodeOrigin> single_origin(uint32_t id, const std::string &name);
+
+std::shared_ptr<CircleNodeOrigin>
+composite_origin(const std::initializer_list<std::shared_ptr<CircleNodeOrigin>> &origins);
```

Comment:
I'm not good at this but I usually seen `initializer_list origins` without `&` for `initializer_list`. maybe because the caller usually called with `{A, B, ...}` without declaring a variable: I can assume why but don't know this is a must or somekind...

Reply:
FYI, https://stackoverflow.com/a/17803511
> It’s passed by value because it’s cheap.

Reply:
@llFreetimell , you didn't add any about my comment

Reply:
@seanshpark 

I thought that key point of your comment was "What is a difference between passing by reference and passing by value in initializer_list?" and https://stackoverflow.com/a/17803511 became an answer. (Frankly speaking, I didn't know that knowledge, until @mhs4670go commented it)

Sorry for misunderstanding :(

Can you let me know what is your main question between following candidates..?
- Should(must) I use `&` in our codes?
- Why did I used `&` in our codes?
- Something else...

Reply:
like other codes, it would be better to remove `&`